qiskit_experiments/data_processing/nodes.py

```diff
@@ -314,61 +286,42 @@ def __init__(self, scale: float = 1.0, validate: bool = True):
         super().__init__(validate)
 
     @abstractmethod
-    def _process(self, datum: np.array, error: Optional[np.array] = None) -> np.array:
+    def _process(self, data: np.ndarray) -> np.ndarray:
         """Defines how the IQ point is processed.
 
-        The dimension of the input datum corresponds to different types of data:
-        - 2D represents average IQ Data.
-        - 3D represents either a single-shot datum or all data of averaged data.
-        - 4D represents all data of single-shot data.
+        The last dimension of the array should correspond to [real, imaginary] part of data.
 
         Args:
-            datum: A N dimensional array of complex IQ points as [real, imaginary].
-            error: A N dimensional array of errors on complex IQ points as [real, imaginary].
+            data: A full data array to process.
 
         Returns:
-            Processed IQ point and its associated error estimate.
+            The data that has been processed.
         """
 
-    def _format_data(self, datum: Any, error: Optional[Any] = None) -> Tuple[Any, Any]:
-        """Check that the IQ data has the correct format and convert to numpy array.
+    def _format_data(self, data: np.ndarray) -> np.ndarray:
+        """Format and validate the input.
+
+        Check that the given data has the correct structure. This method may
+        additionally change the data type, e.g. converting a list to a numpy array.
```

Comment:
This seems a bit strange given that the input is a numpy array.

Reply:
good catch. Done in https://github.com/Qiskit/qiskit-experiments/pull/481/commits/4177cc8132596dd76922728ca0ee08e7be45c54d